### PR TITLE
Make prerelease process work on Windows

### PR DESCRIPTION
**Pull request by rotu**
- Use shutil.which instead of manually finding git in path. This finds git.exe properly and is cleaner.
- Accept CR LF instead of just LF when trying to find version section

### CODE REVIEW

File: src/catkin_pkg/changelog_generator_vcs.py [hunk 1]
```diff
@@ -348,7 +352,6 @@ def get_vcs_client(base_path):
     client_types = [c.type for c in vcs_clients]
     if len(client_types) != len(set(client_types)):
         raise RuntimeError('Multiple vcs clients share the same type: %s' % ', '.join(sorted(client_types)))
-
```
Comment (dirk-thomas) [inline src/catkin_pkg/changelog_generator_vcs.py]: Please revert this unrelated change.

File: src/catkin_pkg/cli/prepare_release.py [hunk 2]
```diff
@@ -194,10 +195,16 @@ def push_changes(base_path, vcs_type, tag_name, dry_run=False):
 
 
 def _find_executable(vcs_type):
-    for path in os.getenv('PATH').split(os.path.pathsep):
-        file_path = os.path.join(path, vcs_type)
-        if os.path.isfile(file_path):
+    try:
+        file_path = shutil.which(vcs_type)
+        if file_path is not None:
             return file_path
```
Comment (dirk-thomas) [inline src/catkin_pkg/cli/prepare_release.py]: Please move the last two lines out of the `try` / `except` block since they can't raise an `AttributeError`

File: src/catkin_pkg/changelog_generator_vcs.py (hunk 1 above)
Comment (rotu) [reply src/catkin_pkg/changelog_generator_vcs.py]: Oops. Done.

File: src/catkin_pkg/cli/prepare_release.py (hunk 2 above)
Comment (rotu) [reply src/catkin_pkg/cli/prepare_release.py]: I switched this over to a better approach. It's more obvious to catch an ImportError

**Comment by dirk-thomas** [comment]
Thanks for the patch.
